gateway/src/config.rs

```diff
@@ -0,0 +1,130 @@
+use std::collections::HashMap;
+
+use figment::{
+    providers::{Env, Format, Serialized, Toml},
+    Figment,
+};
+use serde::{Deserialize, Serialize};
+
+/// A proxy route
+#[derive(Deserialize, Serialize, Clone)]
+pub struct ProxyTarget {
+    /// The request path must start with this prefix for the target to apply.
+    /// If omitted, the target becomes the default. There can only be a single default target.
+    pub prefix: Option<String>,
+    /// The base URL to requests are proxied to (must include the scheme)
+    pub upstream: String,
+    /// Whether requests need authentication to be relayed upstream.
+    /// If true, unauthenticated requests get a 401 Unauthorized response.
+    pub require_auth: bool,
+    /// A list of headers that need to be forwarded. If omitted, all headers are forwarded.
+    /// Omitting this field is not recommended, as it can introduce normalization induced priviledge escalation.
+    pub forwarded_headers: Option<Vec<String>>,
+}
+
+#[derive(Deserialize, Serialize, Clone)]
+pub struct FilesConfig {
+    /// The root folder filesystem path
+    pub root_folder: String,
+    /// When serving a static folder, import the env and modify a file
+    pub import_meta_env: Option<ImportMetaEnv>,
+}
+
+/// Configuration for the import_meta_env feature
+/// This feature will import the environment variables listed in env_vars
+/// and replace the content of the file with the content of the environment
+/// variables
+#[derive(Deserialize, Serialize, Clone)]
+pub struct ImportMetaEnv {
+    pub filename: String,
+    pub env_vars: Vec<String>,
+    pub also_serve_root: bool,
+}
+
+/// Authentication provider
+#[derive(Deserialize, Serialize, Clone)]
+#[serde(tag = "type")]
+pub enum AuthProvider {
+    /// Mocked authentication provider
+    /// This provider will always return the same username
+    Mocked {
+        provider_id: String,
+        username: String,
+        require_login: bool,
+    },
+
+    /// Basic authentication provider
+    /// This provider will check the username and password against a list of
+    /// username/password pairs
+    Bearer {
+        provider_id: String,
+        tokens: HashMap<String, String>,
+    },
+
+    /// OpenID Connect authentication provider
+    /// This provider will redirect the user to the issuer_url to authenticate
+    /// and then redirect the user to the callback_url
+    /// Check the documentation of oidc for more information
+    Oidc {
+        provider_id: String,
+        issuer_url: Box<actix_auth::oidc::IssuerUrl>,
+        post_login_url: Box<actix_auth::oidc::Url>,
+        callback_url: Box<actix_auth::oidc::RedirectUrl>,
+        client_id: String,
+        client_secret: String,
+    },
+}
+
+#[derive(Deserialize, Serialize, Clone)]
```

Comment:
We could add the `deny_unknown_fields`.

```suggestion
#[derive(Deserialize, Serialize, Clone)]
#[serde(deny_unknown_fields)]
```